bundles/org.openhab.core.auth.oauth2client/src/main/java/org/openhab/core/auth/oauth2client/internal/OAuthClientServiceImpl.java

```diff
@@ -400,10 +413,9 @@ private String createNewState() {
     }
 
     @Override
-    public <T extends JsonDeserializer<?>> OAuthClientService withDeserializer(Class<T> deserializerClass) {
+    public OAuthClientService withGsonBuilder(GsonBuilder gsonBuilder) {
```

Comment:
This is not the factory pattern, just a method named similarly, but actually creating a new object once again. This could probably be moved to `OAuthFactory`.